File: 4_Tópicos_avançados/4.8_OpenCV/Ex07/Misturar_Imagens.cpp
// Baseado em https://docs.opencv.org/master/db/df5/tutorial_linux_gcc_cmake.html

#include <stdio.h>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;

int main(int argc, char** argv)
{
	if(argc != 4)
	{
		printf("Modo de uso: %s Nome_Imagem1 Nome_Imagem2 Peso\n",
			argv[0]);
		printf("Peso deve estar entre 0 e 1\n");
		return -1;
	}
	// Ler imagem e guardar em um objeto
	// da classe 'Mat'
	Mat img1 = imread(argv[1]);
	Mat img2 = imread(argv[2]);

	if((img1.rows!=img2.rows) ||
		(img1.cols!=img2.cols) ||
		(img1.channels()!=img2.channels()))
	{
		printf("%s e %s devem ter as mesmas dimensões!\n",
			argv[1], argv[2]);
		return -2;
	}
	
	float p1 = atof(argv[3]);
	if((p1<0.0)||(p1>1.0))
	{
		printf("Peso deve estar entre 0 e 1.\n");
		return -3;
	}
	float p2 = 1.0-p1;
	
	Mat img3;
	addWeighted(img1, p1, img2, p2, 0.0, img3);
	imshow("", img3);
	waitKey(0);
	return 0;
}

File: 4_Dicas/4.2_OpenCV/Ex07/Misturar_Imagens.py
import cv2, numpy, sys

if len(sys.argv)<4:
	print("Modo de uso: " + sys.argv[0] + " Nome_Imagem1 Nome_Imagem2 Peso")
	print("Peso deve estar entre 0 e 1")
	sys.exit(-1)

img1 = cv2.imread(sys.argv[1])
img2 = cv2.imread(sys.argv[2])

rows1,cols1,channels1 = img1.shape
rows2,cols2,channels2 = img2.shape
if (rows1!=rows2) or (cols1!=cols2) or (channels1!=channels2):
	print("%s e %s devem ter as mesmas dimensões!" %
		(sys.argv[1], sys.argv[2]))
	sys.exit(-2)

p1 = float(sys.argv[3])
if (p1<0.0) or (p1>1.0):
	print("Peso deve estar entre 0 e 1.")
	sys.exit(-3)
p2 = 1.0-p1

img3 = cv2.addWeighted(img1,p1,img2,p2,0)
cv2.imshow("", img3)
cv2.waitKey(0)
cv2.destroyAllWindows()